Kleisli.md

```diff
@@ -0,0 +1,169 @@
+autoscale: true
+footer: ![Arrow](arrow-brand-128x128.png) [@47deg](https://twitter.com/47deg) :: [Λrrow](http://arrow-kt.io) :: [http://arrow-kt.io/docs/datatypes/kleisli/](http://arrow-kt.io/docs/datatypes/kleisli/)
+slidenumbers: true
+
+# Kleisli 
+
+__`Kleisli`__ is a data type used in __Λrrow__ to model a transformation 
+
+from one type into another inside a monadic context
+
+---
+
+# Kleisli :: ADT
+
+__`Kleisli`__ represents an arrow from __`<D>`__ to a monadic value __`Kind<F, A>`__.
+
+That means when we create a `Kleisli<Id,Int,Double>` 
+
+we are doing `<Int> -> Id<Double>`
+
+---
+
+# Kleisli :: ADT
+
+Inside the `Kleisli` we specify the transformation.
+
+If we want to transform from the `Int` to the `Id<Double>`
+
+```kotlin
+val doubleIdKleisli = Kleisli { number: Int ->
+  Id.pure(number.toDouble())
+}
+
+val doubleId = doubleIdKleisli.run(1)
+//Id(1.0)
+```
+
+---
+
+# Kleisli :: Local
+
+The __`local`__ function allows doing a conversion inside the `Kleisli` 
```

Comment:
Here we need to show how local it's used to convert two values of different Kleisli types by lifting them to a common dependency.
```kotlin
val k1: Kleisli<ForOption, Int, String> = Kleisli { Some(it.toString()) }
val k2: Kleisli<ForOption, Double, String> = Kleisli { Some(it.toString()) }

data class Config(val n: Int, val d: Double)

val composedToString: Kleisli<ForOption, Config, String> =
  Kleisli.monad<ForOption, Config>().binding {
    val a = k1.local<Config> { it.n }.bind()
    val b = k2.local<Config> { it.d }.bind()
    a + b
  }.fix()
```
This is one of the most important op people have to deal with when using Kelisli's with input dependencies not matching the types, otherwise you can't compose them.